discordbot/OverworldDiscordBot.py

```diff
@@ -0,0 +1,282 @@
+        self.config = None
+        self.is_configured = False
+
+        # function to run when the bot is "ready"
+        @self.event
+        async def on_ready():
+            # tell user in console that the bot is ready
+            _message = "Overworld Discord Bot has started."
+            await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+            # change bot presence to "Watching some ram for shinies"
+            await self.change_presence(activity=discord.Activity(type=discord.ActivityType.watching, name="some ram for shinies"))
+
+        # function to run whenever the start command is called
+        @self.command()
+        async def start(ctx):
+            if not self.is_configured:
+                _message = "Unable to start bot due to missing configuration! Exiting."
+                await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+                return
+            # create reader object
+            try:
+                self.reader = SWSHReader(self.config["IP"])
+                # create and start a thread for reading from the switch
+                self.thread = Thread(target=self.reader_func,args=(ctx,))
+                self.thread.start()
+            except socket.timeout:
+                _message = "Unable to connect to IP {}. Check that the Switch is available.".format(self.config["IP"])
+                await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+
+        # function to run whenever the ping command is called
+        @self.command()
+        async def ping(ctx):
+            # say the bots latency in discord channel
+            await ctx.send(f'Pong! {round(self.latency, 3)}')
+        
+        # function to run whenever stop command is called, stops reader and discord bot
+        @self.command()
+        async def stop(ctx):
+            # tell user in console that we are now stopping
+            _message = "Overworld Discord Bot is stopping."
+            await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+            # if reader was started then close it
+            if self.reader != None:
+                self.reader.close()
+            # tell reader thread that we are done
+            self.thread_running = False
+            # wait for thread to catch up
+            sleep(2)
+            # close discord bot
+            await self.close()
+            _message = "Overworld Discord Bot has stopped."
+            await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+
+    def configure(self, cfg_json):
+        self.config = cfg_json
+        self.is_configured = True
+
+    async def send_discord_event(self, embed, file, channel_id):
+        channel = self.get_channel(int(channel_id))
+        asyncio.run_coroutine_threadsafe(channel.send(embed=embed,file=file),self.loop)
+
+    async def send_discord_msg(self, message, destination=Channels.NoChannel):
```

Comment:
a parameter and filter for whether or not to ping, along with the id of the user/role to ping added to config would be helpful for if you want to log some things but only ping for others. Possibly needs to wait for the filter class in the future

Reply:
Added as a future improvement in the readme.